### PR TITLE
Version 12 and DataModel Fixes

### PR DESCRIPTION
**Pull request by JFranklin3000**
Closes [issue URL]

##### In this PR
- Fixes for Focus Drop in v12
- Modemaster and Pony Roles
- fix for altMode sizes

##### Testing
- Drop focus on sheet in v11 and/or v12 
- Drop Modemaster or pony role on to actor
- validate altModes have a size.

### CODE REVIEW

File: module/sheet-handlers/role-handler.mjs [hunk 1]
```diff
@@ -376,7 +383,6 @@ export async function onLevelChange(actor, newLevel) {
 export async function onRoleDelete(actor, role) {
   const previousLevel = actor.getFlag('essence20', 'previousLevel');
   const focus = getItemsOfType("focus", actor.items);
-
```
Comment (phildominguez) [inline module/sheet-handlers/role-handler.mjs:379]: Revert

File: module/sheet-handlers/role-handler.mjs [hunk 2]
```diff
@@ -163,9 +163,16 @@ export async function focusUpdate(actor, focus, dropFunc) {
...
+  if (game.version < '12') {
+    if (role[0].flags.core.sourceId != attachedRole[0].uuid) {
+      ui.notifications.error(game.i18n.localize('E20.FocusRoleMismatchError'));
+      return false;
+    }
+  } else {
+    if (role[0]._stats.compendiumSource != attachedRole[0].uuid) {
+      ui.notifications.error(game.i18n.localize('E20.FocusRoleMismatchError'));
+      return false;
+    }
```
Comment (phildominguez) [inline module/sheet-handlers/role-handler.mjs:175]:
```suggestion
  const sourceId = game.version < '12'
    ? role[0].flags.core.sourceId
    : role[0]._stats.compendiumSource

  if (sourceId != attachedRole[0].uuid) {
    ui.notifications.error(game.i18n.localize('E20.FocusRoleMismatchError'));
    return false;
  }
```
Not tested

File: module/sheet-handlers/attachment-handler.mjs [hunk 3]
```diff
@@ -272,7 +272,9 @@ export async function _addItemIfUnique(droppedItem, targetItem, entry) {
 */
 export function deleteAttachmentsForItem(item, actor, previousLevel=null) {
   for (const actorItem of actor.items) {
-    const itemSourceId = actor.items.get(actorItem._id).getFlag('core', 'sourceId');
+    const itemSourceId = game.version < 12
```
Comment (phildominguez) [inline module/sheet-handlers/attachment-handler.mjs:275]: foundry.utils.isNewerVersion?

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: module/sheet-handlers/role-handler.mjs (hunk 1 above)
Comment (JFranklin3000) [reply module/sheet-handlers/role-handler.mjs]: reverted

File: module/sheet-handlers/attachment-handler.mjs (hunk 3 above)
Comment (JFranklin3000) [reply module/sheet-handlers/attachment-handler.mjs]: changed to isNewerVersion

**Review by phildominguez — APPROVED** [review]

**Pull request merged**